CCRH/RH850_F1KM_S4/port.c

```diff
@@ -0,0 +1,292 @@
+#define portINT_PRIORITY_LEVEL1             (0x00C1U) /* Level 1 */
+#define portINT_PRIORITY_LEVEL2             (0x00C2U) /* Level 2 */
+#define portINT_PRIORITY_LEVEL3             (0x00C3U) /* Level 3 */
+#define portINT_PRIORITY_LEVEL4             (0x00C4U) /* Level 4 */
+#define portINT_PRIORITY_LEVEL5             (0x00C5U) /* Level 5 */
+#define portINT_PRIORITY_LEVEL6             (0x00C6U) /* Level 6 */
+#define portINT_PRIORITY_LEVEL7             (0x00C7U) /* Level 7 */
+#define portINT_PRIORITY_LEVEL8             (0x00C8U) /* Level 8 */
+#define portINT_PRIORITY_LEVEL9             (0x00C9U) /* Level 9 */
+#define portINT_PRIORITY_LEVEL10            (0x00CAU) /* Level 10 */
+#define portINT_PRIORITY_LEVEL11            (0x00CBU) /* Level 11 */
+#define portINT_PRIORITY_LEVEL12            (0x00CCU) /* Level 12 */
+#define portINT_PRIORITY_LEVEL13            (0x00CDU) /* Level 13 */
+#define portINT_PRIORITY_LEVEL14            (0x00CEU) /* Level 14 */
+#define portINT_PRIORITY_LOWEST             (0x00CFU) /* Level 15 (lowest) */
+
+/*-----------------------------------------------------------*/
+
+/* Counts the interrupt nesting depth. A context switch is only performed
+ * if the nesting depth is 0. */
+volatile BaseType_t xInterruptNesting = 0;
+
+/* Set to 1 to pend a context switch from an ISR. */
+volatile BaseType_t xPortSwitchRequired = pdFALSE;
+
+/* Function to start the first task executing */
+extern void vPortStartFirstTask(void);
+
+/*
+ * Sets up the periodic ISR used for the RTOS tick using the OSTM.
+ * The application writer can define configSETUP_TICK_INTERRUPT() (in
+ * FreeRTOSConfig.h) such that their own tick interrupt configuration is used
+ * in place of prvSetupTimerInterrupt().
+ */
+static void prvSetupTimerInterrupt(void);
+
+#ifndef configSETUP_TICK_INTERRUPT
+
+/* The user has not provided their own tick interrupt configuration so use
+ * the definition in this file (which uses the interval timer). */
+ #define configSETUP_TICK_INTERRUPT()    prvSetupTimerInterrupt()
+#endif                                 /* configSETUP_TICK_INTERRUPT */
+
+/******************************************************************************
+ * Setup the stack of a new task so it is ready to be placed under the
+ * scheduler control. The registers have to be placed on the stack in the
+ * order that the port expects to find them.
+ *
+ * @param[in]  pxTopOfStack  Pointer to top of this task's stack
+ * @param[in]  pxCode        Task function, stored as initial PC for the task
+ * @param[in]  pvParameters  Parameters for task
+ ******************************************************************************/
+StackType_t * pxPortInitialiseStack (StackType_t * pxTopOfStack, TaskFunction_t pxCode, void * pvParameters)
+{
+    /* Simulate the stack frame as it would be created by a context switch
+     * interrupt. */
+    *pxTopOfStack = (StackType_t) vPortEndScheduler;           /* R31 (LP) */
```

Comment:
@TrongNguyenR  I know that it is just semantics but I would like if we use a new function, something like `vFinished` which then asserts.
I know that the current implementation would have the same behavior looking at your implementation of `vPortEndScheduler` function - but it feels incorrect when reading - I do not want my scheduler to stop if one of the tasks exit.

Regardless, I do not think that this is blocking and I am approving the PR. Thank you for taking the time to contribute to FreeRTOS @TrongNguyenR!